server/DataBase/models/user.js

```diff
@@ -0,0 +1,65 @@
+const mongoose = require('mongoose');
+const validator = require('validator');
+const jwt = require('jsonwebtoken');
+const _ = require('lodash');
+const fs = require('fs');
+const bcrypt = require('bcryptjs');
+
+let UserSchema = new mongoose.Schema({
+    name : {
+        type : String,
+        required : true,
+        minlength : 1
+    },
+    email : {
+        type : String,
+        required : true,
+        minlength : 1,
+        trim : true,
+        unique : true,
+        validate : {
+            validator : validator.isEmail,
+            message: '{VALUE} is not a valid email'
+        }
+    },
+    password : {
+        type : String,
+        required: true,
+        minlength : 6
+    },
+    progress : [{
+        course : String,
+        totalQuestions : Number,
+        completedQuestions : {type : Number, default : 0},
+        score : {type : Number, default : 0}
+    }],
+    image: {type : String, default : "images/container-bg.png"}
+});
+
+UserSchema.methods.toJSON = function(){
```

Comment:
explain this why we use this and whats the other method better way?

Reply:
It first converts instance of course model to an object and then return the object as JSON